src/Compilers/Core/Portable/PortableShim.cs

```diff
@@ -34,6 +34,32 @@ namespace Roslyn.Utilities
     /// </summary>
     internal static class PortableShim
     {
+        internal static void Initialize()
+        {
+            Touch(Assembly.Type);
+            Touch(Directory.Type);
+            Touch(Encoding.Type);
+            Touch(Environment.Type);
+            Touch(File.Type);
+            Touch(FileAccess.Type);
+            Touch(FileMode.Type);
+            Touch(FileOptions.Type);
+            Touch(FileShare.Type);
+            Touch(FileStream.Type);
+            Touch(FileVersionInfo.Type);
+            Touch(MemoryStream.Type);
+            Touch(Path.Type);
+            Touch(RuntimeHelpers.Type);
+            Touch(SearchOption.Type);
+            Touch(Thread.Type);
+            Touch(XPath.Extensions.Type);
+        }
+
+        private static void Touch(Type type)
+        {
+            // Do nothing.
```

Comment:
You might want to add a comment explaining this is to trigger the static constructor.